policy/modules/system/systemd.te

```diff
@@ -61,6 +62,14 @@ init_unit_file(systemd_binfmt_unit_t)
 type systemd_conf_t;
 files_config_file(systemd_conf_t)
 
+type systemd_generator_t;
+type systemd_generator_exec_t;
+init_system_domain(systemd_generator_t, systemd_generator_exec_t)
```

Comment:
After further thought, it seems like init_system_domain() should also go into the systemd_generator() interface, so then you have a block like this:
```
type systemd_generator_t;
￼type systemd_generator_exec_t;
systemd_unit_generator(systemd_generator_t, systemd_generator_exec_t)
```
I think we should also rename the interface to `systemd_unit_generator`.

Reply:
Yes, that seems like a reasonable change.  And does make it a bit cleaner.  I have pushed the changes for this.  Let me know if there is anything else that is needed.